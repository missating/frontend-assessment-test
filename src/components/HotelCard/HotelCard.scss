.hotelcard-container {
  --height: 10rem;
  display: flex;
  align-items: center;
  margin-bottom: 1rem;
  width: 100%;

  &__image {
    flex: 0 0 var(--height);
    height: var(--height);
    img {
      width: 100%;
      height: 100%;
      background-color: #cccccc;
    }
  }

  &__details {
    display: flex;
    justify-content: space-between;
    padding: 1rem;
    background-color: #fff;

    h2 {
      color: var(--brand-color);
    }

    p {
      color: rgb(30, 29, 34);
      font-size: 0.7rem;
      line-clamp: 2;
      display: -webkit-box;
      -webkit-line-clamp: 2;
      -webkit-box-orient: vertical;
      overflow: hidden;
      text-overflow: ellipsis;
    }

    &-two {
      display: flex;
      flex-direction: column;
      justify-content: space-between;
    }
  }
}
