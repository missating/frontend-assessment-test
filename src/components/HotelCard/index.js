import React, { useState, useEffect, memo } from 'react'
import StarRating from '../StarRating'
import Reviews from '../Reviews'

import './HotelCard.scss'

const API_URL = 'http://fake-hotel-api.herokuapp.com/api/reviews'

const HotelCard = ({
  image, name, description, price, star, hotelId, city, country, start, end,
}) => {
  const [retried, setRetried] = useState(false);
  const [reviews, setReviews] = useState([])
  const [showReview, setShowReview] = useState(false)
  const [id, setId] = useState(0)

  const handleImgError = (event) => {
    if (!retried) {
      setRetried(true)
      const img = event.target;
      img.src = 'https://res.cloudinary.com/dxayftnxb/image/upload/v1574791542/download_dtr4in.png'
    }
  }

  const toggleReview = () => {
    setShowReview(!showReview)
  }

  useEffect(() => {
    fetch(`${API_URL}?hotel_id=${id}`)
      .then((response) => response.json())
      .then((response) => {
        setReviews(response)
      })
      .catch((error) => {
        throw error
      })
  }, [id])

  return (
    <>
      <article className="hotelcard-container">
        <div className="hotelcard-container__image">
          <img onError={handleImgError} src={image} alt={name} />
        </div>
        <div className="hotelcard-container__details">
          <div className="hotelcard-container__details-one">
            <h2>{name}</h2>
            <p className="hotelcard-container__details-location">
              {city}
              ,
              {' '}
              {' '}
              {country}
            </p>
            <p className="hotelcard-container__details-description">{description}</p>
            <p className="hotelcard-container__details-date">
              Available from
              {' '}
              {start}
              {' '}
              -
              {' '}
              {end}
            </p>
            <StarRating totalStars={star} />
          </div>
          <div className="hotelcard-container__details-two">
            <span>
              <sup>&euro;</sup>
              {price}
            </span>
            <button type="button" className="hotelcard-container__button" onClick={() => { setId(hotelId); toggleReview(); }}>
              Reviews
            </button>
          </div>
        </div>
      </article>
      <div className={showReview ? 'hotelcard-reviews-container' : ''}>
        {
          reviews && reviews.length === 0 && (
            showReview ? (
              <p className="hotelcard-reviews-container__text">
                No Reviews yet
              </p>
            ) : null

          )
        }
        {reviews && reviews.map((data, index) => (
          showReview
            ? (
              <Reviews
                key={data.index}
                name={data.name}
                comment={data.comment}
                positive={data.positive}
              />
            )
            : null
        ))}
      </div>
    </>
  )
}


export default memo(HotelCard)
