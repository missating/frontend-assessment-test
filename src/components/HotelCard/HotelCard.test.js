// react library
import React from 'react';
import { shallow } from 'enzyme';

// components
import HotelCard from './index'

describe('HotelCard Component', () => {
  const initialProps = {
    image: 'exmaple.png',
    name: 'Best Hotel',
    description: 'Best hotel ever',
    price: 500,
    star: 4,
    hotelId: '12345',
    city: 'Lagos',
    country: 'Nigeria',
    start: 'Thu Nov 21 2019',
    end: 'Thu Jul 30 2020',
  }
  it('should be rendered properly', () => {
    // eslint-disable-next-line react/jsx-props-no-spreading
    const wrapper = shallow(<HotelCard {...initialProps} />);
    expect(wrapper.find('div').length).toEqual(5);
    expect(wrapper.find('article').length).toEqual(1)
    expect(wrapper).toMatchSnapshot();
  });
})
