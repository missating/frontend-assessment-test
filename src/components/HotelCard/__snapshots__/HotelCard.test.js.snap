// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HotelCard Component should be rendered properly 1`] = `
<Fragment>
  <article
    className="hotelcard-container"
  >
    <div
      className="hotelcard-container__image"
    >
      <img
        alt="Best Hotel"
        onError={[Function]}
        src="exmaple.png"
      />
    </div>
    <div
      className="hotelcard-container__details"
    >
      <div
        className="hotelcard-container__details-one"
      >
        <h2>
          Best Hotel
        </h2>
        <p
          className="hotelcard-container__details-location"
        >
          Lagos
          ,
           
           
          Nigeria
        </p>
        <p
          className="hotelcard-container__details-description"
        >
          Best hotel ever
        </p>
        <p
          className="hotelcard-container__details-date"
        >
          Available from
           
          Thu Nov 21 2019
           
          -
           
          Thu Jul 30 2020
        </p>
        <starRating
          totalStars={4}
        />
      </div>
      <div
        className="hotelcard-container__details-two"
      >
        <span>
          <sup>
            â‚¬
          </sup>
          500
        </span>
        <button
          className="hotelcard-container__button"
          onClick={[Function]}
          type="button"
        >
          Reviews
        </button>
      </div>
    </div>
  </article>
  <div
    className=""
  />
</Fragment>
`;
